### PR TITLE
HCAP-1297/1282/1283 - testing followup

### PR DESCRIPTION
**Pull request by mmumby**
# Work completed
- Tests for route `/allocations/bulk-allocation`
- tests for new services `createBulkAllocation` and `getAllocationsForSites`
- Updated tests for `getAllPhases` with new queryParam `includeAllocations`
- Cypress tests for bulk allocations 

## Bug fix
- During testing of `createBulkAllocation`, I noticed that the response was being returned before the updates were resolved. Updated the logic to wait for all updates to complete before returning the response. 
- Updated SonarCloud code smells from the previous PR

### Extra
- added a flag to the jest server tests to include colour formatting - This isn't necessary, Its a nice visual,  will remove if vetoed

### CODE REVIEW

**Review by fw-noel — APPROVED** [review]

File: server/services/allocations.ts [hunk 1]
```diff
@@ -88,24 +88,27 @@ export const updateAllocation = async (allocationId: number, allocation, user: H
...
-        existingAllocations.forEach(async (allocation) => {
+        const promises = existingAllocations.map(async (allocation) => {
           const data = {
             allocation: payload.allocation,
             phase_id: payload.phase_id,
             site_id: allocation.site_id,
           };
-          updateResults.push(await updateAllocation(allocation.id, data, user));
+          return await updateAllocation(allocation.id, data, user);
         });
```
[the remark this answers is not in the document]
Comment (rhattie-ey) [reply server/services/allocations.ts:110]: Not sure why Github thinks I wanted to delete the surrounding code, but you get the idea.

File: server/services/allocations.ts [hunk 2]
```diff
@@ -88,24 +88,27 @@ export const updateAllocation = async (allocationId: number, allocation, user: H
  * @param user User performing the operation
  * @returns array of DB operation results
  */
-export const createBulkAllocation = async (payload: bulkAllocationPayload, user: HcapUserInfo) => {
+export const createBulkAllocation = async (payload: BulkAllocationPayload, user: HcapUserInfo) => {
   const updateResults = [];
```
[the remark this answers is not in the document]
Comment (mmumby) [reply server/services/allocations.ts:92]: Are you referring to the []?

File: server/services/allocations.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (mmumby) [reply server/services/allocations.ts:110]: I actually have to refactor this more - It's not doing what I thought it was. It's pushing the promises to the results during every loop of siteIds, which was not the case prior to this update. I'll let you know when its ready for review again.

File: server/tests/phase.e2e.test.js [hunk 3]
```diff
@@ -135,9 +135,17 @@ describe('api e2e tests for /phase', () => {
...
 
-  it('should return phases', async () => {
+  it('should return phases without query param', async () => {
     const header = await getKeycloakToken(healthAuthority);
     const res = await request(app).get(`/api/v1/phase`).set(header);
+    expect(res.body.data[0]).not.toHaveProperty('allocations');
+    expect(res.status).toEqual(200);
+  });
+
+  it('should return phases with allocations with query param `?includeAllocations=true`', async () => {
```
[the remark this answers is not in the document]
Comment (mmumby) [reply server/tests/phase.e2e.test.js:145]: Also the sonarCloud fail is due to adding these additional cases, let's ignore that as these tests are good to have.

**Review by vanics — APPROVED** [review]
Nice work!

**Review by fw-noel — APPROVED** [review]

**Review by rhattie-ey — APPROVED** [review]
Looks great to me now!

**Pull request merged**